adapter/pkg/messaging/notificationListener.go

```diff
@@ -90,71 +100,69 @@ func handleNotification(deliveries <-chan amqp.Delivery, done chan error) {
 // handleAPIEvents to process api related data
 func handleAPIEvents(data []byte, eventType string) {
 	var (
-		apiEvent     APIEvent
-		oldTimeStamp int64
-		indexOfAPI   int
-		isFound      bool
-		newTimeStamp int64 = apiEvent.Event.TimeStamp
+		apiEvent         APIEvent
+		currentTimeStamp int64 = apiEvent.Event.TimeStamp
 	)
 
 	json.Unmarshal([]byte(string(data)), &apiEvent)
-	timeStampList := APIListTimeStamp
-	for apiID, timeStamp := range timeStampList {
-		fmt.Println(apiID, " timeStamp value is", timeStamp)
-		if strings.EqualFold(apiEvent.APIID, apiID) {
-			oldTimeStamp = timeStamp
-		}
-	}
-
-	APIListTimeStamp[apiEvent.APIID] = newTimeStamp
-
-	for i := range APIList {
-		if strings.EqualFold(apiEvent.APIID, string(APIList[i].APIID)) {
-			isFound = true
-			indexOfAPI = i
-			break
-		}
+	// Per each revision, synchronization should happen.
+	if strings.EqualFold(deployAPIToGateway, apiEvent.Event.Type) {
+		go synchronizer.FetchAPIsFromControlPlane(apiEvent.UUID, apiEvent.GatewayLabels)
 	}
 
-	logger.LoggerMsg.Infof("oldTimeStamp: %v , newTimeStamp: %v", oldTimeStamp, newTimeStamp)
-	if isFound && oldTimeStamp < newTimeStamp && strings.EqualFold(removeAPIFromGateway, apiEvent.Event.Type) {
-		deleteAPIFromList(indexOfAPI, apiEvent.APIID)
-	} else if strings.EqualFold(deployAPIToGateway, apiEvent.Event.Type) {
-		// pull API details
-		ID, err := strconv.ParseInt(apiEvent.APIID, 01, 32)
-		if err != nil {
-			logger.LoggerMsg.Errorf("Cannot cast %s to an Integer", apiEvent.APIID)
-		} else {
-			api := resourceTypes.API{APIID: strconv.FormatInt(ID, 10), UUID: apiEvent.UUID,
-				Provider: apiEvent.APIProvider, Name: apiEvent.APIName,
-				Version: apiEvent.APIVersion, Context: apiEvent.APIContext, APIType: apiEvent.APIType,
-				APIStatus: apiEvent.APIStatus, IsDefaultVersion: true, TenantID: apiEvent.TenantID,
-				TenantDomain: apiEvent.Event.TenantDomain, TimeStamp: apiEvent.Event.TimeStamp}
-
-			if apiEvent.Event.Type == "API_CREATE" {
-				subscription.APIList.List = append(subscription.APIList.List, api)
-			} else if apiEvent.Event.Type == "API_UPDATE" {
-				subscription.APIList.List = removeAPI(subscription.APIList.List, apiEvent.APIID)
-				subscription.APIList.List = append(subscription.APIList.List, api)
-			} else if apiEvent.Event.Type == "API_DELETE" {
-				subscription.APIList.List = removeAPI(subscription.APIList.List, apiEvent.APIID)
+	// TODO: (VirajSalaka) Handle API Blocked event
+	if len(apiEvent.GatewayLabels) > 0 {
```

Comment:
remove the if condition as it is handled by the for loop itself.

Reply:
fixed via https://github.com/wso2/product-microgateway/pull/1651/commits/34135e565e0feb892396850af240c35fdc8c4499